ugs-core/src/com/willwinder/universalgcodesender/model/UGSEvent.java

```diff
@@ -176,4 +191,9 @@ public ControllerStatus getControllerStatus() {
     public GcodeState getGcodeState() {
       return gcodeState;
     }
+
+    @Override
+    public String toString() {
+        return ToStringBuilder.reflectionToString(this);
```

Comment:
I've never seen this before, nice trick.

Reply:
I know, what other gems could be hiding in Apache Commons? =)

These are my other favorites:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/HashCodeBuilder.html
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html